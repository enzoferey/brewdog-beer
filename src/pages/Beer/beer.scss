:export {
  backgroundOverflow: #cdcdcd;
}

.main {
  position: relative;
  overflow: hidden;
}

.backIcon {
  position: absolute;
  top: 31px;
  left: 33px;
  color: white;
  font-size: 21px;
  text-decoration: none;

  > img {
    height: 22px;
    vertical-align: text-top;
    margin-right: 15px;
  }
}

.ovale {
  position: absolute;
  top: 0;
  left: 0;
  width: 120%;
  transform: translateX(-8%);
}

.content {
  position: relative;
  z-index: 1;
  width: 90%;
  max-width: 905px;
  margin: 15% auto 0;
}
