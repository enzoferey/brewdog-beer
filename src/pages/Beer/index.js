export { Beer, default } from "./Beer";
