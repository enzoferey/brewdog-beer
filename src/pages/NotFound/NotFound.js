import React from "react";
import PropTypes from "prop-types";
import style from "./not-found.scss";

import { Link } from "react-router-dom";

const NotFound = ({ message }) => (
  <div className={style.main}>
    <h2>{message}</h2>
    <Link to="/">Go to Home</Link>
  </div>
);

NotFound.propTypes = {
  message: PropTypes.string,
};
NotFound.defaultProps = {
  message: "Woops, this page doesn't seem to exist",
};

export default NotFound;
