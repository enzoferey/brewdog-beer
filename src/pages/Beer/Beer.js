import React from "react";
import PropTypes from "prop-types";
import style from "./beer.scss";

import BeerDetails from "components/BeerDetails";
import NotFound from "pages/NotFound";

const Beer = ({ beers, match }) => {
  const urlBeer = match.params.beerName;
  const beer = beers.find(beer => beer.name === urlBeer);

  if (!beer)
    return (
      <NotFound
        message={`Beer "${urlBeer}" doesn't seem to exist, try with another one`}
      />
    );

  const { image_url: image, ...rest } = beer;
  return (
    <div className={style.main}>
      <BeerDetails image={image} {...rest} />
    </div>
  );
};

Beer.propTypes = {
  beers: PropTypes.arrayOf(PropTypes.object),
  match: PropTypes.shape({
    params: PropTypes.shape({
      beerName: PropTypes.string,
    }),
  }),
};

export default Beer;
