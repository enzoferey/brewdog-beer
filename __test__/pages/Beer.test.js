import React from "react";
import { Beer } from "pages/Beer";
import NotFound from "pages/NotFound";

const BeerRouter = withRouter(Beer);

test("renders correctly", () => {
  testSnapshot(
    <BeerRouter
      beers={mocks.beers}
      match={{ params: { beerName: mocks.beers[0].name } }}
    />
  );
});

test("renders NotFound when unkown beer name", () => {
  const wrapper = shallow(
    <BeerRouter
      beers={mocks.beers}
      match={{ params: { beerName: "this beer doesn't exist" } }}
    />
  );

  expect(
    wrapper
      .find(Beer)
      .dive()
      .find(NotFound).length
  ).toBe(1);
});
