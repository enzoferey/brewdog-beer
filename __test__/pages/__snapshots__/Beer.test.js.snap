// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
    }
  }
>
  <div>
    <div>
      <div>
        <svg
          viewBox="0 0 1445.797 391.5"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <linearGradient
              gradientUnits="objectBoundingBox"
              id="linear-gradient"
              x1="0.061"
              x2="1.479"
              y2="1.666"
            >
              <stop
                offset="0"
              />
              <stop
                offset="1"
              />
            </linearGradient>
          </defs>
          <path
            d="M722.9,0c399.246,0,722.9,175.281,722.9,391.5H0C0,175.281,323.653,0,722.9,0Z"
            fill="url(#linear-gradient)"
            transform="translate(1445.797 391.5) rotate(180)"
          />
        </svg>
      </div>
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Back arrow icon"
          src="back-arrow.png"
        />
        Back
      </a>
      <div>
        <div>
          <img
            alt="Beer bottle image"
            src="https://images.punkapi.com/v2/keg.png"
          />
          <h2>
            Buzz
          </h2>
          <h4>
            Alc. 4.5% By Vol.
          </h4>
          <p>
            A light, crisp and bitter IPA brewed with English and American hops. A small batch brewed only once.
          </p>
        </div>
        <div>
          <h2>
            Hops
          </h2>
          <div>
            <div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <span>
                  Fuggles
                </span>
              </div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <img
                  alt=""
                  src="weight.png"
                />
                <span>
                  25 grams
                </span>
              </div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <img
                  alt=""
                  src="time.png"
                />
                <span>
                  start
                </span>
              </div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <img
                  alt=""
                  src="label.png"
                />
                <span>
                  bitter
                </span>
              </div>
              <div>
                <div
                  className="undefined "
                  onClick={[Function]}
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                >
                  <p>
                    Add
                  </p>
                  <div
                    className="undefined "
                    style={
                      Object {
                        "width": "NaN%",
                      }
                    }
                  />
                </div>
              </div>
            </div>
            <div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <span>
                  First Gold
                </span>
              </div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <img
                  alt=""
                  src="weight.png"
                />
                <span>
                  25 grams
                </span>
              </div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <img
                  alt=""
                  src="time.png"
                />
                <span>
                  start
                </span>
              </div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <img
                  alt=""
                  src="label.png"
                />
                <span>
                  bitter
                </span>
              </div>
              <div>
                <div
                  className="undefined "
                  onClick={[Function]}
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                >
                  <p>
                    Add
                  </p>
                  <div
                    className="undefined "
                    style={
                      Object {
                        "width": "NaN%",
                      }
                    }
                  />
                </div>
              </div>
            </div>
            <div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <span>
                  Fuggles
                </span>
              </div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <img
                  alt=""
                  src="weight.png"
                />
                <span>
                  37.5 grams
                </span>
              </div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <img
                  alt=""
                  src="time.png"
                />
                <span>
                  middle
                </span>
              </div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <img
                  alt=""
                  src="label.png"
                />
                <span>
                  flavour
                </span>
              </div>
              <div>
                <div
                  className="undefined undefined"
                  onClick={[Function]}
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                >
                  <p>
                    Add prev
                  </p>
                  <div
                    className="undefined "
                    style={
                      Object {
                        "width": "NaN%",
                      }
                    }
                  />
                </div>
              </div>
            </div>
            <div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <span>
                  First Gold
                </span>
              </div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <img
                  alt=""
                  src="weight.png"
                />
                <span>
                  37.5 grams
                </span>
              </div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <img
                  alt=""
                  src="time.png"
                />
                <span>
                  middle
                </span>
              </div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <img
                  alt=""
                  src="label.png"
                />
                <span>
                  flavour
                </span>
              </div>
              <div>
                <div
                  className="undefined undefined"
                  onClick={[Function]}
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                >
                  <p>
                    Add prev
                  </p>
                  <div
                    className="undefined "
                    style={
                      Object {
                        "width": "NaN%",
                      }
                    }
                  />
                </div>
              </div>
            </div>
            <div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <span>
                  Cascade
                </span>
              </div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <img
                  alt=""
                  src="weight.png"
                />
                <span>
                  37.5 grams
                </span>
              </div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <img
                  alt=""
                  src="time.png"
                />
                <span>
                  end
                </span>
              </div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <img
                  alt=""
                  src="label.png"
                />
                <span>
                  flavour
                </span>
              </div>
              <div>
                <div
                  className="undefined undefined"
                  onClick={[Function]}
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                >
                  <p>
                    Add prev
                  </p>
                  <div
                    className="undefined "
                    style={
                      Object {
                        "width": "NaN%",
                      }
                    }
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <h2>
            Malts
          </h2>
          <div>
            <div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <span>
                  Maris Otter Extra Pale
                </span>
              </div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <img
                  alt=""
                  src="weight.png"
                />
                <span>
                  3.3 kilograms
                </span>
              </div>
              <div>
                <div
                  className="undefined "
                  onClick={[Function]}
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                >
                  <p>
                    Add
                  </p>
                  <div
                    className="undefined "
                    style={
                      Object {
                        "width": "NaN%",
                      }
                    }
                  />
                </div>
              </div>
            </div>
            <div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <span>
                  Caramalt
                </span>
              </div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <img
                  alt=""
                  src="weight.png"
                />
                <span>
                  0.2 kilograms
                </span>
              </div>
              <div>
                <div
                  className="undefined "
                  onClick={[Function]}
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                >
                  <p>
                    Add
                  </p>
                  <div
                    className="undefined "
                    style={
                      Object {
                        "width": "NaN%",
                      }
                    }
                  />
                </div>
              </div>
            </div>
            <div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <span>
                  Munich
                </span>
              </div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <img
                  alt=""
                  src="weight.png"
                />
                <span>
                  0.4 kilograms
                </span>
              </div>
              <div>
                <div
                  className="undefined "
                  onClick={[Function]}
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                >
                  <p>
                    Add
                  </p>
                  <div
                    className="undefined "
                    style={
                      Object {
                        "width": "NaN%",
                      }
                    }
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <h2>
            Methods
          </h2>
          <div>
            <div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <div>
                  <div>
                    <div>
                      <span>
                        1
                      </span>
                    </div>
                    <img
                      alt=""
                      src="temperature.png"
                    />
                    <span>
                      64ºC
                    </span>
                    <span>
                      5s
                    </span>
                  </div>
                </div>
              </div>
              <div
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <img
                  alt=""
                  src="fermentation.png"
                />
                <span>
                  19ºC
                </span>
              </div>
              <div
                style={
                  Object {
                    "flex": 2,
                  }
                }
              >
                <p>
                  -
                </p>
              </div>
              <div>
                <div
                  className="undefined "
                  onClick={[Function]}
                  style={
                    Object {
                      "background": undefined,
                    }
                  }
                >
                  <p>
                    Add
                  </p>
                  <div
                    className="undefined "
                    style={
                      Object {
                        "width": "100%",
                      }
                    }
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
